plugins/out_pgsql/pgsql_loadlib.h

```diff
@@ -0,0 +1,163 @@
+    CONNECTION_GSS_STARTUP,     /* Negotiating GSSAPI. */
+    CONNECTION_CHECK_TARGET,    /* Checking target server properties. */
+    CONNECTION_CHECK_STANDBY    /* Checking if server is in standby mode. */
+} ConnStatusType;
+
+typedef enum
+{
+    PGRES_EMPTY_QUERY = 0,      /* empty query string was executed */
+    PGRES_COMMAND_OK,           /* a query command that doesn't return anything was executed properly by the backend */
+    PGRES_TUPLES_OK,            /* a query command that returns tuples was executed properly by the backend, PGresult contains the result tuples */
+    PGRES_COPY_OUT,             /* Copy Out data transfer in progress */
+    PGRES_COPY_IN,              /* Copy In data transfer in progress */
+    PGRES_BAD_RESPONSE,         /* an unexpected response was recv'd from the backend */
+    PGRES_NONFATAL_ERROR,       /* notice or warning message */
+    PGRES_FATAL_ERROR,          /* query failed */
+    PGRES_COPY_BOTH,            /* Copy In/Out data transfer in progress */
+    PGRES_SINGLE_TUPLE,         /* single tuple from larger resultset */
+    PGRES_PIPELINE_SYNC,        /* pipeline synchronization point */
+    PGRES_PIPELINE_ABORTED      /* Command didn't run because of an abort
+                                                                                                                        * earlier in a pipeline */
+} ExecStatusType;
+
+typedef struct pg_result PGresult;
+typedef struct pg_conn PGconn;
+
+// Declare all symbols loaded from pqlib.dll
+
+typedef int (*PQconsumeInputP)(PGconn *);
+PQconsumeInputP PQconsumeInput;
+
+typedef ConnStatusType (*PQstatusP)(const PGconn *);
+PQstatusP PQstatus;
+
+typedef PGresult* (*PQgetResultP)(PGconn *);
+PQgetResultP PQgetResult;
+
+typedef ExecStatusType (*PQresultStatusP)(const PGresult *);
+PQresultStatusP PQresultStatus;
+
+typedef char* (*PQerrorMessageP)(const PGconn *);
+PQerrorMessageP PQerrorMessage;
+
+typedef void (*PQclearP)(PGresult*);
+PQclearP PQclear;
+
+typedef void (*PQfinishP)(PGconn*);
+PQfinishP PQfinish;
+
+typedef PGconn* (*PQsetdbLoginP)(const char*, const char*, const char*, const char*, const char*, const char*, const char*);
+PQsetdbLoginP PQsetdbLogin;
+
+typedef int (*PQsetnonblockingP)(PGconn*, int);
+PQsetnonblockingP PQsetnonblocking;
+
+typedef int (*PQisBusyP)(PGconn*);
+PQisBusyP PQisBusy;
+
+typedef char* (*PQescapeIdentifierP)(PGconn*, const char*, size_t);
+PQescapeIdentifierP PQescapeIdentifier;
+
+typedef void (*PQfreememP)(void*);
+PQfreememP PQfreemem;
+
+typedef PGresult* (*PQexecP)(PGconn*, const char*);
+PQexecP PQexec;
+
+typedef void (*PQresetP)(PGconn*);
+PQresetP PQreset;
+
+typedef char* (*PQescapeLiteralP)(PGconn*, const char*, size_t);
+PQescapeLiteralP PQescapeLiteral;
+
+typedef int (*PQsendQueryP)(PGconn*, const char*);
+PQsendQueryP PQsendQuery;
+
+typedef int (*PQflushP)(PGconn*);
+PQflushP PQflush;
+
+#endif
+
+struct flb_pgsql_config {
```

Comment:
This is not strict to loadlib.h or anything related to win32, so this should not be here

Reply:
Fixed.
`loadPqDll()` had a param `struct flb_pgsql_config *ctx` initially and I think this was generating a circular dependency issue.
It now uses directly a `struct flb_output_instance*` param since it's needed only to call `flb_plg_warn()`.